Library/OcCpuLib/OcCpuLib.c

```diff
@@ -171,10 +307,15 @@ ScanIntelProcessor (
     //
     if (Cpu->Model >= CPU_MODEL_NEHALEM) {
       PerfStatus.Uint64 = AsmReadMsr64 (MSR_IA32_PERF_STATUS);
-      Cpu->CurBusRatio = (UINT8) (PerfStatus.Bits.State >> 8U);
       PlatformInfo.Uint64 = AsmReadMsr64 (MSR_NEHALEM_PLATFORM_INFO);
       Cpu->MinBusRatio = (UINT8) PlatformInfo.Bits.MaximumEfficiencyRatio;
       Cpu->MaxBusRatio = (UINT8) PlatformInfo.Bits.MaximumNonTurboRatio;
+
+      if (Cpu->Model <= CPU_MODEL_WESTMERE) {
```

Comment:
This condition is very likely wrong for Sandy Bridge CPUs, as Intel names their CPUs in a non-linear way. Could you make some function that will handle the nuance? I get a feeling the IsSandyOrIvy code should be generalised with this function. Perhaps, we need some internal linear generation mapping in OcSupportPkg, just like Lilu's currentGeneration: https://github.com/acidanthera/Lilu/blob/master/Lilu/Sources/kern_cpu.cpp.